resources/qml/UserProfile.qml

```diff
@@ -33,13 +34,33 @@ ApplicationWindow {
             onClicked: TimelineManager.openImageOverlay(TimelineManager.timeline.avatarUrl(userid), TimelineManager.timeline.data.id)
         }
 
-        Label {
+        TextInput {
+            id: displayUsername
+            readOnly: !profile.isUsernameEditingAllowed
             text: profile.displayName
-            fontSizeMode: Text.HorizontalFit
             font.pixelSize: 20
             color: TimelineManager.userColor(profile.userid, colors.window)
             font.bold: true
             Layout.alignment: Qt.AlignHCenter
+            selectByMouse: true
+
+            Keys.priority: Keys.BeforeItem
+            Keys.onReturnPressed: profile.changeUsername(displayUsername.text)
+
+            ImageButton {
+                anchors.leftMargin: 5
+                anchors.left: displayUsername.right
+                anchors.verticalCenter: displayUsername.verticalCenter
+                image: profile.isUsernameEditingAllowed ? ":/icons/icons/ui/checkmark.png" : ":/icons/icons/ui/edit.png"
+
+                onClicked: {
+                    if(profile.isUsernameEditingAllowed) {
+                        profile.changeUsername(displayUsername.text)
+                    }else{
+                        profile.allowUsernameEditing(true)
```

Comment:
I think you should keep the state of if you are currently editing the username to the qml part alone instead of making it a property of the Profile. You should also disallow editing the username, if the user is not yourself.